lib/queue.js

```diff
@@ -62,30 +66,68 @@ Queue.prototype.push = function (entry) {
 }
 
 /**
- * flush to socket server
+ * flush to collector server
  */
 Queue.prototype.flush = function () {
   debug('[%s] sending (%d)', chalk.yellow('agent'), this.alf.har.log.entries.length)
 
+  var flushId = uuid.v4()
+
   // construct HTTP mesasge body
   var postData = JSON.stringify(this.alf)
+  dataDebug('[%s] postData %s', chalk.yellow('agent data'), postData)
 
   // immediatly reset entries object
   this.alf.har.log.entries = []
+  var opts = this.opts
+  opts.userAgent = this.userAgent
+  this.postToCollector(flushId, postData, opts, this.opts.limits.retry)
+}
 
-  var client = this.opts.ssl ? https : http
+Queue.prototype.flushInProgress = function (set, flushId) {
```

Comment:
I'm biased, but https://github.com/SGrondin/bottleneck is made for that, plus you can adjust the level of concurrency as needed.